fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordvalidation/RecordValidationTest.java

```diff
@@ -0,0 +1,408 @@
+            assertFalse(valueValidatorResult.isValid());
+            assertEquals(RecordValueValidator.CODE_DESERIALIZE_ERROR, valueValidatorResult.getErrorCode());
+
+            commit(context);
+        }
+    }
+
+    /**
+     * A test that corrupts the version of the records and ensures it is handled.
+     *
+     * @param formatVersion the version format
+     */
+    @ParameterizedTest(name = "testValidateRecordCorruptVersion [formatVersion = {0}]")
+    @MethodSource("formatVersions")
+    void testValidateRecordCorruptVersion(int formatVersion) throws Exception {
+        final RecordMetaDataHook hook = ValidationTestUtils.getRecordMetaDataHook(true);
+        List<FDBStoredRecord<Message>> result = saveRecords(true, formatVersion, hook);
+        // Mess up the splits
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            // corrupt the two records
+            byte[] key = ValidationTestUtils.getSplitKey(store, result.get(0).getPrimaryKey(), -1);
+            final byte[] value = new byte[] {1, 2, 3, 4, 5};
+            store.ensureContextActive().set(key, value);
+
+            key = ValidationTestUtils.getSplitKey(store, result.get(1).getPrimaryKey(), -1);
+            store.ensureContextActive().set(key, value);
+
+            commit(context);
+        }
+
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            RecordValidator valueValidator = new RecordValueValidator(store);
+            // Currently, we don't handle corrupt versions.
+            Exception ex = assertThrows(ExecutionException.class, () -> valueValidator.validateRecordAsync(result.get(0).getPrimaryKey()).get());
+            assertTrue(ex.getCause() instanceof UnknownValidationException);
+            ex = assertThrows(ExecutionException.class, () -> valueValidator.validateRecordAsync(result.get(1).getPrimaryKey()).get());
+            assertTrue(ex.getCause() instanceof UnknownValidationException);
+
+            commit(context);
+        }
+    }
+
+    private static void validateRecordVersion(final FDBRecordStore store, final Tuple primaryKey, final @Nonnull String expectedVersionValidationCode) throws InterruptedException, ExecutionException {
+        RecordValidator versionValidator = new RecordVersionValidator(store);
+        final RecordValidationResult versionValidatorResult = versionValidator.validateRecordAsync(primaryKey).get();
+        if (expectedVersionValidationCode.equals(RecordValidationResult.CODE_VALID)) {
+            assertTrue(versionValidatorResult.isValid());
+        } else {
+            assertFalse(versionValidatorResult.isValid());
+        }
+        assertEquals(expectedVersionValidationCode, versionValidatorResult.getErrorCode());
```

Comment:
All of this is duplicated in `validateRecordValue`, and could be extracted into a helper.

Reply:
Done.

Reply:
Done.